src/main/java/net/syscon/elite/repository/jpa/model/BedAssignmentHistory.java

```diff
@@ -0,0 +1,68 @@
+package net.syscon.elite.repository.jpa.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Data
+@Builder
+@EqualsAndHashCode(exclude = {"bedAssignmentHistoryPK", "offenderBooking"}, callSuper = false)
+@ToString(exclude = {"bedAssignmentHistoryPK", "offenderBooking"})
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "BED_ASSIGNMENT_HISTORIES")
+public class BedAssignmentHistory extends AuditableEntity {
+
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @Embeddable
+    public static class BedAssignmentHistoryPK implements Serializable {
+        @Column(name = "OFFENDER_BOOK_ID", nullable = false)
+        private Long offenderBookingId;
+
+        @Column(name = "BED_ASSIGN_SEQ", nullable = false)
+        private Integer sequence;
+    }
+
+    @EmbeddedId
+    private BedAssignmentHistoryPK bedAssignmentHistoryPK;
+
+    @ManyToOne
+    @JoinColumn(name = "OFFENDER_BOOK_ID", insertable = false, updatable = false)
+    private OffenderBooking offenderBooking;
```

Comment:
This was required to handle the foreign key on offender booking id, which is part of the composite key and so cannot handle the join.